src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/TokenCertificatesApiController.java

```diff
@@ -0,0 +1,95 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.restapi.openapi;
+
+import ee.ria.xroad.common.conf.serverconf.model.CertificateType;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+import org.niis.xroad.restapi.converter.CertificateDetailsConverter;
+import org.niis.xroad.restapi.exceptions.ErrorDeviation;
+import org.niis.xroad.restapi.openapi.model.CertificateDetails;
+import org.niis.xroad.restapi.service.CertificateAlreadyExistsException;
+import org.niis.xroad.restapi.service.ClientNotFoundException;
+import org.niis.xroad.restapi.service.GlobalConfService;
+import org.niis.xroad.restapi.service.KeyNotFoundException;
+import org.niis.xroad.restapi.service.TokenCertificateService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * certificates api
+ */
+@Controller
+@RequestMapping("/api")
+@Slf4j
+@PreAuthorize("denyAll")
+public class TokenCertificatesApiController implements TokenCertificatesApi {
+    public static final String ERROR_INVALID_CERT_UPLOAD = "invalid_cert_upload";
+    public static final String ERROR_INVALID_CERT = "invalid_cert";
+
+    private final TokenCertificateService tokenCertificateService;
+    private final CertificateDetailsConverter certificateDetailsConverter;
+
+    @Autowired
+    public TokenCertificatesApiController(TokenCertificateService tokenCertificateService,
+            CertificateDetailsConverter certificateDetailsConverter) {
+        this.tokenCertificateService = tokenCertificateService;
+        this.certificateDetailsConverter = certificateDetailsConverter;
+    }
+
+    @Override
+    @PreAuthorize("hasAnyAuthority('IMPORT_AUTH_CERT', 'IMPORT_SIGN_CERT')")
```

Comment:
This would mean XROAD_REGISTRATION_OFFICER can import auth cert, which should not be allowed.

It's tricky since we don't know the certificate type until at TokenCertificateService.

One option would be to check for authorities in TokenCertificateService. Maybe something close to this example: https://stackoverflow.com/a/3021529/1469083

Or you can refactor, and create/change a method so that it has certificate type as a parameter. Annotate this method with proper security annotations. It may involve breaking the principle "only authorize at controller layer" but it has enough justification.

Adding test for the authorization logic would be good. I did something a bit similar work (but easier) in get-csr-parameters implementation: 
- https://github.com/nordic-institute/X-Road/blob/XRDDEV-693-get-certificate-profile-rebased/src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/KeysApiController.java#L119-L125
- https://github.com/nordic-institute/X-Road/blob/XRDDEV-693-get-certificate-profile-rebased/src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/openapi/KeysApiControllerTest.java#L138-L143

Reply:
This has been addressed in the [TokenCertificateService](https://github.com/nordic-institute/X-Road/pull/268/commits/e241e058684f68a6be0def5f7fd946942748773f#diff-8f67256a64f49c079c774afed4998fe3R107-R114)